src/connectors/ledger.js

```diff
@@ -0,0 +1,114 @@
+import Web3 from "web3";
+import createLedgerSubprovider from "@ledgerhq/web3-subprovider";
+import TransportU2F from "@ledgerhq/hw-transport-u2f";
+import ProviderEngine from "web3-provider-engine";
+import WebsocketSubprovider from 'web3-provider-engine/subproviders/websocket'
+import { AbstractConnector } from '@web3-react/abstract-connector' 
+
+
+/**
+ * An implementation of a LedgerConnector for web3-react, based on the original
+ * `@web3-react/ledger-connector`.
+ * 
+ * Some differences:
+ * 
+ * 1. The original doesn't expose the LedgerJS client API. 
+ *    We will probably want access to this in future, eg. signing BTC transactions
+ * 
+ * 2. The original doesn't work with event subscriptions, as it assumes a HTTP RPC 
+ *    endpoint. Event subscriptions use `eth_subscribe`, which HttpProvider does not 
+ *    implement out-of-the-box. There are some packages, such as Metamask's 
+ *    eth-json-rpc-filters, which will implement a middleware to achieve this. Assuming 
+ *    a Websocket provider is simpler for our case.
+ */
+export class LedgerConnector extends AbstractConnector {
+  constructor({
+    chainId,
+    url,
+    pollingInterval,
+    requestTimeoutMs,
+    accountFetchingConfigs,
+    baseDerivationPath
+  }) {
+    super({
+      supportedChainIds: [chainId]
```

Comment:
To what extent does the issue with chain ids above break our ability to interact with our internal testnet, out of curiosity?

Reply:
Just swapping out the current subproviders with ones provided by 0x - they provided more consistent exceptions, that I wasn't hellbent on reimplementing.

Reply:
They may not support the larger chainID. We'll see what we can do.